auth/auth-server/src/telemetry/sources/mod.rs

```diff
@@ -43,6 +40,49 @@ impl QuoteResponse {
 
         quote_amt / base_amt
     }
+
+    /// Calculates the estimated gas cost in USDC
+    pub fn gas_cost(&self, usdc_per_gas: f64) -> f64 {
+        let gas_total = self.gas as f64;
+        gas_total * usdc_per_gas
+    }
+
+    /// Calculates the net output value of a trade, accounting for gas costs.
+    pub fn output_value_net_of_gas(&self, usdc_per_gas: f64, side: OrderSide) -> f64 {
+        // Get decimal corrected amounts
+        let base_token = Token::from_addr(&self.base_mint);
+        let quote_token = Token::from_addr(&self.quote_mint);
+
+        let base_amt = base_token.convert_to_decimal(self.base_amount);
+        let quote_amt = quote_token.convert_to_decimal(self.quote_amount);
+
+        // Get gas cost in USDC
+        let usdc_gas_cost = self.gas_cost(usdc_per_gas);
+
+        // Subtract gas cost from net output value
+        match side {
+            OrderSide::Sell => quote_amt - usdc_gas_cost,
+            OrderSide::Buy => {
+                let usdc_per_base = self.price();
+                let gas_cost_in_base = usdc_gas_cost / usdc_per_base;
+                base_amt - gas_cost_in_base
+            },
+        }
+    }
+}
+
+/// Converts the `AtomicMatchApiBundle` into a `QuoteResponse`.
+impl From<&AtomicMatchApiBundle> for QuoteResponse {
+    fn from(bundle: &AtomicMatchApiBundle) -> Self {
+        Self {
+            quote_mint: bundle.match_result.quote_mint.clone(),
+            base_mint: bundle.match_result.base_mint.clone(),
+            quote_amount: bundle.match_result.quote_amount,
+            base_amount: bundle.match_result.base_amount,
+            gas: DEFAULT_GAS_ESTIMATION,
```

Comment:
why don't we actually estimate the gas cost of the `settlement_tx` in the `AtomicMatchApiBundle`?

Reply:
good point, we now use the estimate if it exists